src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,35 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.Arrays;
+import java.util.concurrent.RecursiveAction;
+
+public class MergeSortAction extends RecursiveAction {
+
+    private static final int THRESHOLD = 3;
+    private final int[] array;
+    private final int length;
+    private final int start;
 
     public MergeSortAction(int[] array) {
+        this(array, 0, array.length);
+    }
+
+    private MergeSortAction(int[] array, int length, int start) {
+        this.array = array;
+        this.length = length;
+        this.start = start;
+    }
+
+    @Override
+    protected void compute() {
+        if (length - start <= THRESHOLD) {
+            Arrays.sort(array);
```

Comment:
The sort operation is being applied to the entire array, rather than the subarray defined by 'start' and 'length'. This will result in incorrect behavior when sorting subarrays since the whole array will be sorted instead of the intended portion.